yb-voyager/src/srcdb/oracle.go

```diff
@@ -37,12 +37,33 @@ func (ora *Oracle) GetTableRowCount(tableName string) int64 {
 	log.Infof("Querying row count of table %q", tableName)
 	err := ora.db.QueryRow(query).Scan(&rowCount)
 	if err != nil {
-		utils.ErrExit("Failed to query row count of %q: %s", tableName, err)
+		utils.ErrExit("Failed to query %q for row count of %q: %s", query, tableName, err)
 	}
 	log.Infof("Table %q has %v rows.", tableName, rowCount)
 	return rowCount
 }
 
+func (ora *Oracle) GetTableApproxRowCount(tableProgressMetadata *utils.TableProgressMetadata) int64 {
+	var approxRowCount sql.NullInt64 // handles case: value of the row is null, default for int64 is 0
+	var query string
+	if !tableProgressMetadata.IsPartition {
```

Comment:
Looks like you will also need to accept the `IsPartition` and `ParentTableName` too as arguments.